crates/validation/src/storage_mapping.rs

```diff
@@ -1,13 +1,49 @@
 use crate::errors::Error;
 
 use super::{indexed, reference};
+use models::Store;
 use superslice::Ext;
 
 pub fn walk_all_storage_mappings(
     storage_mappings: &[tables::StorageMapping],
     errors: &mut tables::Errors,
 ) {
     for m in storage_mappings {
+        for store in m.stores.iter() {
+            // TODO: it seems like we should also be calling `walk_name` for the bucket and prefix, right?
```

Comment:
I left this as a TODO because we already aren't performing this validation, and I not entirely certain that our regex validation even matches what's accepted by cloud providers. TBH I really can't think of any security concerns related to this, but please LMK if you can. I think we should introduce that validation, but I'd like to decouple that from this PR if possible, so I can check to make sure that all our existing values would even pass such validation.